app/services/authorizer.rb

```diff
@@ -12,8 +12,9 @@ def can?(permission, subject = nil)
       user.permissions.where(:name => permission).present?
     else
       return true if user.admin?
-      collection = @cache[subject.class.to_s][permission] ||= find_collection(subject.class, :permission => permission)
-      collection.include?(subject)
+      @cache["#{subject.class.to_s}-#{permission}-#{subject.id}"] ||=
+        find_collection(subject.class, :permission => permission).
+        where(:id => subject.id)
```

Comment:
Yeah you're right - although these queries are considerably faster now that I wonder whether the cache is even necessary now. I did it this way because storing the collection in the cache would mean having to retrieve collection (without `where`), which was the bottleneck.

I'll see how can I signal to this method that I don't want to get `find_collection`, maybe with an option hash as you said !